dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +249,54 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
         &hKey,
         nullptr /* lpdwDisposition */));
 
-    // Retrieve DisplayName and IconUri
-    // - DisplayName: Retrieve from Shell. If not specified, fall back to filename.
-    // - Icon: Retrieve from Shell. If it's not the case or the file extension is unsupported, then throw.
-    winrt::com_ptr<IPropertyStore> propertyStore;
-    wil::unique_hwnd hWindow{ GetConsoleWindow() };
-    THROW_IF_FAILED(SHGetPropertyStoreForWindow(hWindow.get(), IID_PPV_ARGS(propertyStore.put())));
-
     // Retrieve the display name
     std::wstring displayName{};
+    displayName = SetDisplayNameBasedOnProcessName();
 
-    wil::unique_prop_variant propVariantDisplayName;
-    // Do not throw in case of failure. We can use the filepath approach below as fallback to set a DisplayName.
-    LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
-    if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
-    {
-        displayName = propVariantDisplayName.pwszVal;
-    }
+    // Retrieve the icon
+    std::wstring iconFilePath{};
 
-    if (displayName.length() == 0)
+    wil::unique_hwnd hWindow{ GetConsoleWindow() };
```

Comment:
This doesn't throw right if ConSoleWindow is not present? That would be really bad.

Reply:
https://docs.microsoft.com/en-us/windows/console/getconsolewindow - doesn't say anything about throwing. It returns null if there is no handle.

Reply:
Documentation indicates that it would return null: https://docs.microsoft.com/en-us/windows/console/getconsolewindow

Reply:
Ok looks like our responses crossed. This should be good then.